### PR TITLE
Change one-hot constraints to use discrete variables

### PR DESCRIPTION
**Pull request by akoszegi**
Now that we have discrete variables in cqm I thought we could use them here. 

Is there a way to implement the partition size equality constraint using symbolic math + discrete variables?

### CODE REVIEW

**Comment by vgoliber** [comment]
If using discrete variables then there's no need for double-indexing and the problem can be entirely reformulated.

**Comment by arcondello** [comment]
Yeah, in the future we can make this even nicer with https://github.com/dwavesystems/dimod/issues/921 and https://github.com/dwavesystems/dimod/issues/996

File: demo.py [hunk 1]
```diff
@@ -121,13 +121,14 @@ def build_cqm(G, k):
...
-    v = [[Binary(f'v_{i},{k}') for k in partitions] for i in G.nodes]   
+    v = [[Binary(f'v_{i},{k}') for k in partitions] for i in G.nodes]
+    labels = [[f'v_{i},{k}' for k in partitions] for i in G.nodes]
 
     # One-hot constraint: each node is assigned to exactly one partition
     print("\nAdding one-hot constraints...")
     for n in G.nodes:
         # print("\nAdding one-hot for node", n)
-        cqm.add_constraint(quicksum(v[n]) == 1, label='one-hot-node-{}'.format(n)) 
+        cqm.add_discrete(labels[n], label=f"one-hot-node-{n}")
```
Comment (hhtong) [inline demo.py:131]: Alternatively, you could could combine this line with line 125:
```
for n in G.nodes:
  cqm.add_discrete([f'v_{n},{k}' for k in partitions], label=f"one-hot-node-{n}")
```

**Review by hhtong — APPROVED** [review]

File: demo.py (hunk 1 above)
Comment (akoszegi) [reply demo.py]: Thanks for the suggestion! I used it instead of what I had :)

**Pull request merged**